docs/errors.md

```diff
@@ -0,0 +1,231 @@
+# Error Class Hierarchy
+
+**Title**: Error Class Hierarchy
+
+**Authors**: Warren James, Andy Mina
+
+**Advisory Group**: Daria Purdue, Eric Adum, Neal Beeken
+
+### Contents
+
+- [Introduction](#Introduction)
+- [Errors](#errors)
+  - [`MongoError`](#MongoError)
+  - [`MongoDriverError`](#MongoDriverError)
+    - [`MongoLogicError`](#MongoLogicError)
+    - [`MongoRuntimeError`](#MongoRuntimeError)
+  - [`MongoNetworkError`](#MongoNetworkError)
+  - [`MongoServerError`](#MongoServerError)
+  - [`MongoSystemError`](#MongoSystemError)
+- [Test Plan](#Test-Plan)
+  - [`MongoLogicError`](#MongoLogicError-1)
+    - [`MongoInvalidArgumentError`](#MongoInvalidArgumentError-1)
+    - [`MongoMissingCredentialsError`](#MongoMissingCredentialsError-1)
+  - [`MongoRuntimeError`](#MongoRuntimeError-1)
+    - [`MongoNotConnectedError`](#MongoNotConnectedError-1)
+    - [`MongoServerClosedError`](#MongoServerClosedError-1)
+    - [`MongoStreamClosedError`](#MongoStreamClosedError-1)
+    - [`MongoTopologyClosedError`](#MongoTopologyClosedError-1)
+    - [`MongoCursorExhaustedError`](#MongoCursorExhaustedError-1)
+  - [`MongoNetworkError`](#MongoNetworkError-1)
+    - [`MongoNetworkTimeoutError`](#MongoNetworkTimeoutError-1)
+
+# Errors
+
+All errors are derived from the `MongoError` class which should **never** be instantiated.
+There are four main error classes which stem from `MongoError`: `MongoDriverError`, `MongoNetworkError`, `MongoServerError`, and `MongoSystemError`.
+
+## `MongoError`
+
+The base class from which all errors in the Node driver subclass.
+`MongoError` should **never** be be directly instantiated.
+![(MongoError hierarchy tree)](charts/imgs/MongoError.svg)
+
+Children of `MongoError` include:
+
+- [`MongoDriverError`](#MongoDriverError)
+- [`MongoNetworkError`](#MongoNetworkError)
+- [`MongoServerError`](#MongoServerError)
+- [`MongoSystemError`](#MongoSystemError)
+
+## `MongoDriverError`
+
+This class represents errors which originate in the driver itself or in the user's use of the driver.
+This class should **never** be directly instantiated.
+Its children are the main classes of errors that most users will interact with: [**`MongoLogicError`**](#MongoLogicError) and [**`MongoRuntimeError`**](#MongoRuntimeError).
+
+### `MongoLogicError`
```

Comment:
I find that I'm not a huge fan of this name because "logic" is quite vague, would something like `MongoAPIError` be better?

Reply:
@emadum I think `MongoAPIError` definitely does a better job of immediately directing the user where to look so I’ve updated `MongoLogicError`.